### PR TITLE
Convert all `tap` tests to `jest`, drop EOL Node v10

### PR DESCRIPTION
**Pull request by dotkas**
Removed `tap` dependency and converted all tests to `jest.`

Dropped support for EOL NodeJS version 10. 12 and 14 still here until CLI removes it.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: 2.1
 
 orbs:
-  win: circleci/windows@2.4.0
+  win: circleci/windows@5.0.0
```
Comment (orsagie) [inline .circleci/config.yml:4]: Looks like cli still supports 2.4.1, so we should also make sure not to break that: https://github.com/snyk/cli/blob/master/.circleci/config.yml#L5

File: .circleci/config.yml [hunk 2]
```diff
@@ -111,37 +116,39 @@ workflows:
       - lint:
           name: Lint
           context: nodejs-install
-          node_version: '14'
+          node_version: '16.20'
```
Comment (orsagie) [inline .circleci/config.yml:119]: Looks like CLI tests from v12, so lets add those back in https://github.com/snyk/cli/blob/295a14789a0b55d9f1ad71607b8655520566aa09/.github/workflows/smoke-tests.yml#L20

File: .circleci/config.yml (hunk 1 above)
Comment (dotkas) [reply .circleci/config.yml:4]: That version of the orb has a bug in `nvm` so I need to upgrade it, but it only manifests itself on `prettier` so it should not cause the CLI to fail.

File: .circleci/config.yml (hunk 2 above)
Comment (dotkas) [reply .circleci/config.yml:119]:
```
#!/bin/bash -eo pipefail
npm test

> snyk-nuget-plugin@ test /root/snyk-nuget-plugin
> npm run lint && npm run unit-test


> snyk-nuget-plugin@ lint /root/snyk-nuget-plugin
> prettier --check "./lib/**/*.ts" && eslint -c .eslintrc.js "./lib/**/*"

prettier requires at least version 14 of Node, please upgrade
```

Comment (dotkas) [reply .circleci/config.yml:119]: Guess I can live with earlier versions of dev-deps for a while. But this should REALLY be fixed in CLI.

**Review by orsagie — APPROVED** [review]

**Pull request merged**